### PR TITLE
Topk verif

### PR DESCRIPTION
**Pull request by samsja**
## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html).

### CODE REVIEW

File: python/sglang/srt/managers/io_struct.py [hunk 1]
```diff
@@ -48,13 +48,20 @@ class GenerateReqInput:
...
     # If return logprobs, the start location in the prompt for returning logprobs.
     # By default, this value is "-1", which means it will only return logprobs for output tokens.
     logprob_start_len: Optional[Union[List[int], int]] = None
     # If return logprobs, the number of top logprobs to return at each position.
     top_logprobs_num: Optional[Union[List[int], int]] = None
+    
+    # Whether to return hidden states. work the same as return_logprob.
+    return_hidden_states: Optional[Union[List[bool], bool]] = None
+    # If return hidden states, the number of top hidden states to return at each position.
+    top_hidden_states_num: Optional[Union[List[int], int]] = None
```
Comment (Jackmin801) [inline python/sglang/srt/managers/io_struct.py:63]: Its actually not clear to me what this means. Is it the layer? Is it values?

File: python/sglang/srt/layers/logits_processor.py [hunk 2]
```diff
@@ -124,6 +155,22 @@ def __init__(
...
+        if self.do_tensor_parallel_all_gather:
+            pruned_states = tensor_model_parallel_all_gather(pruned_states)
+
+        if logits_metadata.top_hidden_states_nums is None:
+            raise NotImplementedError("Extend mode with return_hidden_states is not supported yet.")
+
+        topk = logits_metadata.top_hidden_states_nums
+
+
+        topk_states =  pruned_states.abs().topk(topk, dim=1)
```
Comment (samsja) [inline python/sglang/srt/layers/logits_processor.py]: @Jackmin801 here

File: python/sglang/srt/layers/logits_processor.py [hunk 3]
```diff
@@ -124,6 +155,23 @@ def __init__(
...
+            raise NotImplementedError("Extend mode with return_hidden_states is not supported yet.")
+
+        topk = logits_metadata.top_hidden_states_nums
+
+
+        if topk == -1:
+            return pruned_states, torch.arange(pruned_states.size(1), device=pruned_states.device)
+        else:
+            topk_states =  pruned_states.topk(topk, dim=1)
+            return topk_states.values, topk_states.indices
```
Comment (samsja) [inline python/sglang/srt/layers/logits_processor.py:173]: @Jackmin801
